cpp/src/parquet/stream_reader.h

```diff
@@ -0,0 +1,159 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PARQUET_STREAM_READER_H
+#define PARQUET_STREAM_READER_H
+
+#include <array>
+#include <chrono>
+#include <cstdint>
+#include <cstring>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parquet/api/reader.h"
+#include "parquet/stream_writer.h"
+
+namespace parquet {
+
+// A class for reading Parquet files using an output stream type API.
+//
+// The values given must be of the correct type i.e. the type must
+// match the file schema exactly otherwise a ParquetException will be
+// thrown.
+//
+// The user must explicitly advance to the next row using the
+// EndRow() function or EndRow input manipulator.
+//
+// Currently there is no support for optional or repeated fields.
+//
+class PARQUET_EXPORT StreamReader {
+ public:
+  StreamReader() = default;
+
+  explicit StreamReader(std::unique_ptr<ParquetFileReader>& File_Reader);
```

Comment:
I commented on the implementation, make this a raw pointer and clarify owernship isn't transferred, or remove the reference and have users do an explicit move.

Reply:
Ok reference removed so user must be explicit move.